Bearing/BearingChapterBuilder.cpp

```diff
@@ -51,15 +52,1161 @@ Uint16 CBearingChapterBuilder::GetMaxLevel() const
+	*pPara << symbol(TIMES) << Sub2(_T("(G"), _T("min")) << _T(" + ") << Sub2(_T("G"), _T("max")) << _T(")");
+	*pPara << symbol(TIMES) << _T("S") << Super(_T("2")) << _T(" = 3") << symbol(TIMES) << _T("(");
+	*pPara << E.SetValue(Gmin) << _T(" + ") << E.SetValue(Gmax) << _T(")") << symbol(TIMES);
+	*pPara << s << Super(_T("2")) << _T(" = ") << E.SetValue(EcA) << rptNewLine;
+	if (max_comp_strain_check == true)
+	{
+		*pPara << symbol(RIGHT_SINGLE_ARROW) << stress.SetValue(tl_stress) << _T("/");
+		*pPara << E.SetValue(EcA) << _T(" = ") << tl_stress / EcA << _T(" < 0.07 ") << RPT_PASS;
+	}
+	else
+	{
+		*pPara << symbol(RIGHT_SINGLE_ARROW) << stress.SetValue(tl_stress) << _T("/");
+		*pPara << E.SetValue(EcA) << _T(" = ") << tl_stress / EcA << _T(" > 0.07 ") << RPT_FAIL;
+	}
+	*pPara << _T(" (14.7.6.3.3-1)");
+
+	pSubHeading = new rptParagraph(rptStyleManager::GetSubheadingStyle());
+	(*pChapter) << pSubHeading;
+	*pSubHeading << _T("Initial Dead Load Deflection:");
+	pPara = new rptParagraph;
+	(*pChapter) << pPara;
+
+	*pPara << Sub2(symbol(delta), _T("DL-initial")) << _T(" = ") << Sub2(_T("P"), _T("DL")) << symbol(TIMES) << Sub2(_T("h"), _T("ri"));
+	*pPara << _T("/A/") << Sub2(_T("E"), _T("c")) << _T(" = ");
+	*pPara << force.SetValue(dl) << symbol(TIMES) << length.SetValue(total_elastomer_thickness);
+	*pPara << _T("/") << _T("/") << area.SetValue(a) << _T("/") << E.SetValue(EcA) << _T(" = ");
+	*pPara << length.SetValue(deltaDLiA) << _T(" (AASHTO Comm. C14.7.5.3.6-1)");
+
+
+	pSubHeading = new rptParagraph(rptStyleManager::GetSubheadingStyle());
+	(*pChapter) << pSubHeading;
+	*pSubHeading << _T("Instantaneous Live Load Deflection:");
+	pPara = new rptParagraph;
+	(*pChapter) << pPara;
+
+	*pPara << Sub2(symbol(delta), _T("LL")) << _T(" = ") << Sub2(_T("P"), _T("LL")) << symbol(TIMES) << Sub2(_T("h"), _T("ri"));
+	*pPara << _T("/A/") << Sub2(_T("E"), _T("c")) << _T(" = ");
+	*pPara << force.SetValue(ll) << symbol(TIMES) << length.SetValue(total_elastomer_thickness);
+	*pPara << _T("/") << _T("/") << area.SetValue(a) << _T("/") << E.SetValue(EcA) << _T(" = ");
+	*pPara << length.SetValue(deltaLLiA) << _T(" (AASHTO Comm. C14.7.5.3.6-1)");
+
+
+
+}
+
 rptChapter* CBearingChapterBuilder::Build(const std::shared_ptr<const WBFL::Reporting::ReportSpecification>& pRptSpec,Uint16 level) const
 {
-   rptChapter* pChapter = new rptChapter;
-   rptParagraph* pPara;
 
-   pPara = new rptParagraph;
+
+   CEAFApp* pApp = EAFGetApp();
+   const WBFL::Units::IndirectMeasure* pDispUnits = pApp->GetDisplayUnits();
+   INIT_UV_PROTOTYPE(rptLengthUnitValue, length, pDispUnits->ComponentDim, true);
+   INIT_UV_PROTOTYPE(rptAreaUnitValue, area, pDispUnits->Area, true);
+   INIT_UV_PROTOTYPE(rptForceUnitValue, force, pDispUnits->GeneralForce, true);
+   INIT_UV_PROTOTYPE(rptStressUnitValue, stress, pDispUnits->Stress, true);
+   INIT_UV_PROTOTYPE(rptStressUnitValue, E, pDispUnits->ModE, true);
+
+
+   const auto brg{ m_pDoc->GetBearing().first };
```

Comment:
Should be `const auto&` otherwise the bearing object is copied

Reply:
this has been updated.